### PR TITLE
small ce queue cleaning

### PR DESCRIPTION
**Pull request by sns-seb**
This PR:
* removes `ReportProcessingScheduler`
* moves queue related classes to their own package

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnable.java [hunk 1]
```diff
@@ -18,42 +18,49 @@
...
+      return queue.peek();
     } catch (Exception e) {
       LOG.error("Failed to pop the queue of analysis reports", e);
-      return;
     }
+    return Optional.absent();
+  }
 
+  private void executeTask(CeTask task) {
     // TODO delegate the message to the related task processor, according to task type
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnable.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* BLOCKER [Catch Exception instead of Throwable.](https://github.com/SonarSource/sonarqube/blob/9d8d0cedfe9e77045565932fe1c6c468ba9b921a/server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnable.java#L69) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)

**Comment by sns-seb** [comment]
@simonbrandhof just rebased this PR on master
good for review

**Comment by simonbrandhof** [comment]
Can't the classes ```PurgeActivities```, ```ComputationTempFolderProvider``` and ```ComputeEngineProcessingModule``` be moved to this new package ```queue``` too ? The package is currently about producer and consumer of the queue.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Complete the task associated to this TODO comment.](https://github.com/SonarSource/sonarqube/blob/eaa0bdb6787b5d6bf4cea87c0dfd443df3f367ac/server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnable.java#L70) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* BLOCKER [Catch Exception instead of Throwable.](https://github.com/SonarSource/sonarqube/blob/eaa0bdb6787b5d6bf4cea87c0dfd443df3f367ac/server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnable.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)

**Comment by sns-seb** [comment]
> Can't the classes ```PurgeActivities```, ```ComputationTempFolderProvider``` and ```ComputeEngineProcessingModule``` be moved to this new package ```queue``` too ? The package is currently about producer and consumer of the queue.

not necessarily in the `queue` package, but I'll move them all to sub packages

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Complete the task associated to this TODO comment.](https://github.com/SonarSource/sonarqube/blob/288964db4c035dc49ee32e7b2114102a6307f426/server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnableImpl.java#L70) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* BLOCKER [Catch Exception instead of Throwable.](https://github.com/SonarSource/sonarqube/blob/288964db4c035dc49ee32e7b2114102a6307f426/server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnableImpl.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)

**Comment by sns-seb** [comment]
@simonbrandhof I have moved classes out of `computation` package and did some cleaning in modules, WDYT?

**Comment by simonbrandhof** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Complete the task associated to this TODO comment.](https://github.com/SonarSource/sonarqube/blob/7463093c64cb231356a56e69870c2a725ac78da8/server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnableImpl.java#L70) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* BLOCKER [Catch Exception instead of Throwable.](https://github.com/SonarSource/sonarqube/blob/7463093c64cb231356a56e69870c2a725ac78da8/server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnableImpl.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Complete the task associated to this TODO comment.](https://github.com/SonarSource/sonarqube/blob/f5ccdcbb3b781c8e430ed26a30c79ce9ad9ad70c/server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnableImpl.java#L70) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* BLOCKER [Catch Exception instead of Throwable.](https://github.com/SonarSource/sonarqube/blob/f5ccdcbb3b781c8e430ed26a30c79ce9ad9ad70c/server/sonar-server/src/main/java/org/sonar/server/computation/queue/CeWorkerRunnableImpl.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)

**Pull request merged**